src/outputs/restart.cpp

```diff
@@ -17,30 +17,582 @@
+    pmb = pm->pblock;
+    i = 0;
+    while (pmb != nullptr) {
+      tmpLoc[i++] = pmb->loc.lx1;
+      tmpLoc[i++] = pmb->loc.lx2;
+      tmpLoc[i++] = pmb->loc.lx3;
+      pmb = pmb->next;
+    }
+    WRITEH5SLABI64("loc.lx123", tmpLoc, gBlocks, local_start, local_count, global_count,
+                   property_list);
+    delete[] tmpLoc;
+
+    // (LOC.)level, GID, LID, cnghost, gflag
+    n = 5;
+    auto *tmpID = new int[num_blocks_local * n];
+    local_count[1] = global_count[1] = n;
+    local_count[0] = num_blocks_local;
+    global_count[0] = max_blocks_global;
+    pmb = pm->pblock;
+    i = 0;
+    while (pmb != nullptr) {
+      tmpID[i++] = pmb->loc.level;
+      tmpID[i++] = pmb->gid;
+      tmpID[i++] = pmb->lid;
+      tmpID[i++] = pmb->cnghost;
+      tmpID[i++] = pmb->gflag;
+      pmb = pmb->next;
+    }
+    WRITEH5SLABI32("loc.level-gid-lid-cnghost-gflag", tmpID, gBlocks, local_start,
+                   local_count, global_count, property_list);
+    delete[] tmpID;
+  }
+
+  // close locations tab
+  H5Gclose(gBlocks);
+
+  // write variables
+
+  // write variables
+  // create persistent spaces
+  local_count[1] = nx3;
+  local_count[2] = nx2;
+  local_count[3] = nx1;
+  local_count[4] = 1;
+
+  global_count[1] = nx3;
+  global_count[2] = nx2;
+  global_count[3] = nx1;
+  global_count[4] = 1;
+
+  hid_t local_DSpace = H5Screate_simple(5, local_count, NULL);
+  hid_t global_DSpace = H5Screate_simple(5, global_count, NULL);
+
+  // while we could do this as n variables and load all variables for
+  // a block at one time, this is memory-expensive.  I think it is
+  // well worth the multiple iterations through the blocks to load up
+  // one variable at a time.  Besides most of the time will be spent
+  // writing the HDF5 file to disk anyway...
+  // If I'm wrong about this, we can always rewrite this later.
+  // Sriram
+
+  const hsize_t varSize = nx3 * nx2 * nx1;
+
+  auto ciX = ContainerIterator<Real>(
+      pm->pblock->real_containers.Get(),
+      {parthenon::Metadata::Independent, parthenon::Metadata::Restart}, true);
+  for (auto &vwrite : ciX.vars) { // for each variable we write
+    const std::string vWriteName = vwrite->label();
+    hid_t vLocalSpace, vGlobalSpace;
+    pmb = pm->pblock;
+    const hsize_t vlen = vwrite->GetDim(4);
+    local_count[4] = global_count[4] = vlen;
+    Real *tmpData = new Real[varSize * vlen * num_blocks_local];
+
+    // create spaces if required
+    if (vlen == 1) {
+      vLocalSpace = local_DSpace;
+      vGlobalSpace = global_DSpace;
+    } else {
+      vLocalSpace = H5Screate_simple(5, local_count, NULL);
+      vGlobalSpace = H5Screate_simple(5, global_count, NULL);
+    }
+
+    // load up data
+    hsize_t index = 0;
+    pmb = pm->pblock;
+    while (pmb != nullptr) { // for every block
+      auto ci = ContainerIterator<Real>(
+          pmb->real_containers.Get(),
+          {parthenon::Metadata::Independent, parthenon::Metadata::Restart}, true);
```

Comment:
Don't want to impose this on you now. But I think my preferred solution to this would be to have a step when variables are created which adds the restart flag if they're independent. Then this line can just be `{Parthenon::Metadata::Restart}`.

Food for thought for later. Shouldn't block this MR.

Reply:
I'm not sure I agree with the philosophy.  But am not going to address it here.

Reply:
Yeah that's fine. Like I said, don't want to block this MR.